packages/medusa/src/services/event-bus.ts

```diff
@@ -147,10 +167,15 @@ export default class EventBusService {
       throw new Error("Subscriber must be a function")
     }
 
-    if (this.observers_.get(event)?.length) {
-      const index = this.observers_.get(event)?.indexOf(subscriber)
-      if (index !== -1) {
-        this.observers_.get(event)?.splice(index as number, 1)
+    const subscribers = this.observers_.get(event)
```

Comment:
suggestion(non blocking): We retrieve the observerDescriptors, there is to much confusion with the wording as it seems to be a subscribers but in fact it is not ahah. I think the previous proposal could help on clarifying